aten/src/ATen/quantized/Quantizer.cpp

```diff
@@ -126,10 +127,18 @@ Tensor quantize_tensor(Tensor rtensor, Tensor qtensor, double scale, int64_t zer
   qparams.scale = scale;
   qparams.zero_point = zero_point;
   qparams.precision = CHAR_BIT * sizeof(typename T::underlying);
-  fbgemm::Quantize<typename T::underlying>(/*src=*/rd,
-                             /*dst=*/qd,
-                             /*len=*/rtensor.numel(),
-                             /*qparams=*/qparams);
+  int num_tasks = at::get_num_threads();
+  at::parallel_for(0, num_tasks, 1, [&](int64_t begin, int64_t end) {
```

Comment:
could this be done inside fbgemm?

Reply:
By design, fbgemm does not create any threads internally and this way we can make use of any threading system (OpenMP, TBB etc). By keeping it here, pytorch decides how many threads to create and which threading system to use.